tools/observability/resource-utilization-dash.json

```diff
@@ -0,0 +1,632 @@
+            "type": "prometheus",
+            "uid": "prometheus"
+          },
+          "editorMode": "code",
+          "expr": "max by (modelName, label_determined_ai_experiment_id)( DCGM_FI_DEV_MEM_COPY_UTIL * on(pod) group_left(label_determined_ai_experiment_id, label_determined_ai_task_type) kube_pod_labels{label_determined_ai_task_type = \"TRIAL\"})",
+          "instant": false,
+          "legendFormat": "__auto",
+          "range": true,
+          "refId": "A"
+        }
+      ],
+      "title": "GPU Memory Utilization by Experiment ID and GPU Model",
+      "type": "timeseries"
+    },
+    {
+      "datasource": {
+        "type": "prometheus",
+        "uid": "prometheus"
+      },
+      "description": "Average GPU Utilization (amount of individual GPU used) by task type over time.",
+      "fieldConfig": {
+        "defaults": {
+          "color": {
+            "mode": "palette-classic"
+          },
+          "custom": {
+            "axisBorderShow": false,
+            "axisCenteredZero": false,
+            "axisColorMode": "text",
+            "axisLabel": "",
+            "axisPlacement": "auto",
+            "barAlignment": 0,
+            "drawStyle": "line",
+            "fillOpacity": 0,
+            "gradientMode": "none",
+            "hideFrom": {
+              "legend": false,
+              "tooltip": false,
+              "viz": false
+            },
+            "insertNulls": false,
+            "lineInterpolation": "linear",
+            "lineWidth": 1,
+            "pointSize": 5,
+            "scaleDistribution": {
+              "type": "linear"
+            },
+            "showPoints": "auto",
+            "spanNulls": true,
+            "stacking": {
+              "group": "A",
+              "mode": "none"
+            },
+            "thresholdsStyle": {
+              "mode": "off"
+            }
+          },
+          "mappings": [],
+          "thresholds": {
+            "mode": "absolute",
+            "steps": [
+              {
+                "color": "green",
+                "value": null
+              },
+              {
+                "color": "red",
+                "value": 80
+              }
+            ]
+          }
+        },
+        "overrides": []
+      },
+      "gridPos": {
+        "h": 8,
+        "w": 12,
+        "x": 0,
+        "y": 16
+      },
+      "id": 8,
+      "options": {
+        "legend": {
+          "calcs": [],
+          "displayMode": "list",
+          "placement": "bottom",
+          "showLegend": true
+        },
+        "tooltip": {
+          "mode": "single",
+          "sort": "none"
+        }
+      },
+      "targets": [
+        {
+          "datasource": {
+            "type": "prometheus",
+            "uid": "prometheus"
+          },
+          "editorMode": "code",
+          "expr": "avg by (label_determined_ai_task_type)( DCGM_FI_DEV_GPU_UTIL * on(pod) group_left(label_determined_ai_task_type) kube_pod_labels )",
```

Comment:
![image](https://github.com/determined-ai/determined/assets/31451191/f78394c1-9bc9-4b3c-abc2-6b1ebf8dd108)

I noticed if I launch a 1 slot notebook and don't do anything to the notebook it has a value of 0 on this chart. I think it has a value of 0 because the notebook isn't using the GPU because I haven't done anything yet. Despite the Notebook showing 0 it is still taking up the full gpu from a cluster management perspective. No one else can use these gpus while this notebook is running.

I wonder if DCGM is the right metric to look at for this chart. I wonder if instead we want to know how much k8s gpu requests the pod is holding instead.

Maybe the DCGM metrics would be better for a task level page?

Reply:
I tried refining the plots a bit ... relying more on a count of GPUs "in use" than the utilization (how much of a gpu is in use). 

There's two ways we can measure whether a GPU is "in use".
- count GPUs reporting a `DCGM_FI_DEV_GPU_UTIL` metric, aka `count by (modelName) (DCGM_FI_DEV_GPU_UTIL)`
- count GPUs reporting **non zero** `DCGM_FI_DEV_GPU_UTIL`, aka `count by (modelName) (DCGM_FI_DEV_GPU_UTIL > 0)`

The first one shows a smoother line ... there won't be a gap if a GPU is temporarily not used because a pod has ended and a new one is transitioning from pending to running.

The second one better indicates that a notebook is using a GPU node, but not the GPU itself. 

I switched based on what seemed more useful. LMK what you think.

Reply:
On second thought, the query `count by (modelName) (DCGM_FI_DEV_GPU_UTIL)` wouldn't go back down to zero after all the trials completed. Not going to use the first one ... I don't think it's doing what I think it's doing (or reporting is wonky)

Reply:
If I launch a notebook and don't do anything won't the reported use be 0? At least that is what I saw in this.

It seems weird to me we are asking DCGM for this when this is almost more a scheduler concern. Does ```kube_pod_resource_request``` return GPU information? https://kubernetes.io/docs/reference/instrumentation/metrics/

Is there a way to gpu requests from kube state metrics somehow?